main.cpp

```diff
@@ -0,0 +1,419 @@
+#include <windows.h>
+#include <map>
+#include "TextureHolder.h"
+#include "Square.h"
+#include "Piece.h"
+
+bool move(Square* slctd, Square* trgt, sf::UdpSocket& sckt, sf::IpAddress& IP, unsigned short& PORT, sf::Packet& pck);
+
+int main()
+{	
+	//kod debugowania = 1337
+	bool turn=false, gamestarted=false, iswhite = true;
+	sf::Packet pack;
+	sf::UdpSocket socket;
+	sf::IpAddress serveradress;
+	unsigned short serverport;
+	int serverconnectionstatus = 1, id, ctx, cty;
+	float tx, ty; //koordy targetu do update
+
+
+
+
+	std::cout << "Server IP: ";
+	std::cin >> serveradress;
+	std::cout << "Server Port: ";
+	std::cin >> serverport;
+
+
+	if (serverport != 1337) { // debug - wpisa� w pole "port" 1337 aby przejsc do inicjalizacji gry
+		if (socket.bind(socket.getLocalPort()) != sf::Socket::Done) {
+			std::cout << "Failed to bind a socket!" << std::endl;
+			exit(4);
+		}
+
+		pack << serverconnectionstatus;
+		if (socket.send(pack, serveradress, serverport) != sf::Socket::Done) {
+			std::cout << "Failed to send connection initialization packet!" << std::endl;
+			exit(2);
+		}
+		pack.clear();
+		if (socket.receive(pack, serveradress, serverport) != sf::Socket::Done) {
+			std::cout << "Failed to recieve initialization packet!" << std::endl;
+			exit(2);
+		}
+		serverconnectionstatus = 0;
+		pack >> serverconnectionstatus >> iswhite;
+		std::cout << "CONNSTATUS: " << serverconnectionstatus << std::endl;
+		std::cout << std::boolalpha;
+		std::cout << "Is white? " << iswhite << std::endl;
+
+		if (serverconnectionstatus != 2) {
+			std::cout << "Connection not stablilized!" << std::endl;
+			exit(3);
+		}
+
+		std::cout << "Waiting for 2nd player" << std::endl;
+		pack.clear();
+		if (socket.receive(pack, serveradress, serverport) != sf::Socket::Done) {
+			std::cout << "Failed to recieve initialization packet!" << std::endl;
+			exit(2);
+		}
+		pack >> gamestarted;
+		if (gamestarted) 
+			std::cout << "Everything is ready!" << std::endl;
+
+		else {
+			std::cout << "2nd player didn't come, terminating..." << std::endl;
+			exit(69);
+		}
+
+}
+	
+	//---------------------------------------------------------------------------------------------------------
+	
+
+	//inicjalizacja okna
+	TextureHolder TXHOLDER;
+	sf::RenderWindow _window(sf::VideoMode(800, 800	), "Chess", sf::Style::Titlebar | sf::Style::Close);
+	_window.setFramerateLimit(75);
+	sf::Event event;
+	sf::Vector2f mouse;
+
+
+	//inicjalizacja szachownicy
+	Square* sqrs[8][8];
+	sf::Color* color = new sf::Color;
+	*color = sf::Color::White;
+
+	for (int i = 0; i < 8; i++) {
+		if (*color == sf::Color::Blue)
+			*color = sf::Color::White;
+
+		else
+			*color = sf::Color::Blue;
+
+		for (int j = 0; j < 8; j++) {
+			if (*color == sf::Color::Blue)
+				*color = sf::Color::White;
+
+			else
+				*color = sf::Color::Blue;
+			Square* field = new Square(0+100*i,0+100*j, *color);
+			sqrs[i][j] = field;
+		}
+	}
+	delete color;
+	
+	//inicjalizacja bierek
+	Piece* pcs[32];
```

Comment:
To też nie powinien być pointer - gdy klasa jest właścicielem, to może przechowywać obiekty 'normalnie'.